core/src/toga/app.py

```diff
@@ -827,7 +827,12 @@ def show_cursor(self) -> None:
 
     @property
     def current_window(self) -> Window | None:
-        """Return the currently active window."""
+        """The window currently in active focus.
+
+        On macOS, when a :any:`Dialog` is dismissed, the window from which the dialog
+        was initiated becomes the :any:`App.current_window`, regardless of whether the
+        :any:`App.current_window` property was modified.
```

Comment:
I'm not convinced this needs to be documented. Yes, it's a platform quirk; but "macOS behaves like macOS" isn't news. If we want to get technical, I'm going to bet that GTK has some pretty quirky behavior here too, because focus management is a window manager concern. The WM we're using for test purposes returns focus to the old current window - but a "focus follows mouse" WM almost certainly won't.

Reply:
I agree with you, so I've removed the note.